app/models/submission_folder_renames.rb

```diff
@@ -0,0 +1,64 @@
+class SubmissionFolderRenames
+  include ActiveModel::Model
+
+  validates :path_new, presence: true
+  attr_accessor :submission, :directory, :renamed_at, :path_old, :path_new, :renamed_by_id
+  delegate :term, :students, :exercise, to: :submission
+
+  def save
+    if valid?
+      if allowed_new_name?
+        submission_assets = @submission.submission_assets.where(path: path_old).
+          or(@submission.submission_assets.where("path like ?", "#{path_old}/%"))
+
+        submission_assets.each do |asset|
+          new_path = asset.path.sub(File.basename(path_old), path_new)
+          asset.path = new_path
+        end
+
+        success = submission_assets.all? do |asset|
+          asset.valid? && asset.save
+        end
+
+        if success
+          return true
+        end
+      end
+    end
+    return false
+  end
+
+  def allowed_new_name?
+    return !illegal_new_name?
```

Comment:
Ruby automatically returns the value of the last expression. Only use `return` when necessary.

```suggestion
    !illegal_new_name?
```